content/news/2024-02-03-bevy-0.13/index.md

```diff
@@ -872,6 +872,50 @@ let layout = render_device.create_bind_group_layout(
 );
 ```
 
+## Unload rendering assets from RAM
+
+<div class="release-feature-authors">authors: @JMS55, @mockersf, @brianreavis</div>
+
+Meshes and the textures used to define their materials take up a ton of memory:
+in many games, memory usage is the biggest limitation on the resolution and polygon count of the game!
+Moreover, actually transferring that data from the CPU (and RAM) to the GPU (and VRAM) can be a real performance bottleneck,
+especially if you're doing it from scratch every frame.
+
+In the [initial PR] by @JMS55, Bevy added the ability to unload this data from the CPU, once it has been succesfully tranferred to the GPU.
+However, unloading the data from the CPU [can result in bugs], and make it harder to actually inspect or tweak that data from your other game systems.
+As a result, this behavior is currently off by default.
+
+To configure this behavior for your asset, set the [`RenderAssetUsages`] field when constructing your raw asset type such as `Image` or `Mesh`,
+which is a bitflag type that allows you to specify whether to retain the data in the main (CPU) world, the render (GPU) world or both.
+
+@brianreavis [later refined] this API, and used it to ensure that texture atlases and font atlases only extract data that's actually in use
+to the GPU, rather than wasting work sending _all_ possible images or characters to the GPU every frame.
+Neat!
+
+[initial PR]: https://github.com/bevyengine/bevy/pull/10520
+[can result in bugs]: https://github.com/bevyengine/bevy/pull/11212
+[`RenderAssetUsages`]: https://dev-docs.bevyengine.org/bevy/render/render_asset/struct.RenderAssetUsages.html
+[later refined]: https://github.com/bevyengine/bevy/pull/11399
+
+## Better batching through smarter sorting
+
+<div class="release-feature-authors">authors: @Elabajaba</div>
+
+One of the core techniques used to speed up rendering is to draw many similar objects together at the same time.
+In this case, Bevy was already using a technique called "batching", which allows us to combine multiple similar operations together,
+reducing the number of expensive draw calls (instructions to the GPU) that are being made.
+
+However, our strategy for defining these batches was far from optimal.
+Previously, we were sorting by distance to the camera, and _then_ checking if multiple of the same meshes were adjacent to each other in that sorted list.
+On realistic scenes, this is unlikely to find many candidates for merging!
+
+Following [PR #11671] however, we first sort by pipeline (effectively the type of material being used), and then by mesh identity.
+This strategy results in much better batching, improving overall FPS by double digit percentages on realistic scenes!
+
+![A graph showing batching improvements. FPS improved by at least 20% in all cases tested.](better_batching.png)
+
+[PR #11671]: https://github.com/bevyengine/bevy/pull/11671
```

Comment:
It would be helpful to mention to compare the `pipeline_sort` vs `main` is what the reader should be looking for.

It took me an embarrassingly long time to realise I wasn't supposed to be comparing the things next to eachother 😅 

Ideally the different tests would be colour coded or something to help the viewer know which ones to compare, I think the sorting by raw fps is unhelpful.

Reply:
Yeah I think I'm going to redo the graph quick.